### ISSUE TITLE
Plugin not working with Redmine 5.0.x

### ISSUE THREAD

**Issue by draschba** [ncoders/redmine_local_avatars#30]
Trying to install the plugin within a Redmine 5.0.x leads to error:

```
The Gemfile's dependencies are satisfied
rake aborted!
LoadError: cannot load such file -- local_avatars
/usr/local/bundle/gems/zeitwerk-2.5.4/lib/zeitwerk/kernel.rb:35:in `require'
/usr/local/bundle/gems/zeitwerk-2.5.4/lib/zeitwerk/kernel.rb:35:in `require'
/usr/src/redmine/plugins/redmine_local_avatars/init.rb:45:in `<top (required)>'
/usr/src/redmine/lib/redmine/plugin_loader.rb:31:in `load'
/usr/src/redmine/lib/redmine/plugin_loader.rb:31:in `run_initializer'
/usr/src/redmine/lib/redmine/plugin_loader.rb:108:in `each'
/usr/src/redmine/lib/redmine/plugin_loader.rb:108:in `block in load'
```

Trying to fix this issue according to the discussion in the redmine board https://www.redmine.org/boards/2/topics/67113 didn't help by changing the requirement pathes to the lib directory in init.rb. 

Any ideas?

**Comment by Kamp11** [comment]
Changed init.rb:
```
#require 'local_avatars'
require File.dirname(__FILE__) + '/lib/local_avatars.rb'

# patches to Redmine
#require "account_controller_patch.rb"
require File.dirname(__FILE__) + '/lib/account_controller_patch.rb'
#require "application_helper_avatar_patch.rb"
require File.dirname(__FILE__) + '/lib/application_helper_avatar_patch.rb'
#require "my_controller_patch.rb"
require File.dirname(__FILE__) + '/lib/"my_controller_patch.rb'
#require "users_avatar_patch.rb"   # User model
require File.dirname(__FILE__) + '/lib/users_avatar_patch.rb'
#require "users_controller_patch.rb"
require File.dirname(__FILE__) + '/lib/"users_controller_patch.rb'
#require "users_helper_avatar_patch.rb"  # UsersHelper
require File.dirname(__FILE__) + '/lib/users_helper_avatar_patch.rb"'

# hooks
#require 'redmine_local_avatars/hooks'
require File.dirname(__FILE__) + '/lib/redmine_local_avatars/hooks.rb'
```

Now the error looks like this:
```
rails aborted!
Zeitwerk::NameError: expected file /opt/redmine/plugins/redmine_local_avatars/lib/local_avatars.rb to define constant LocalAvatars, but didn't
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/callbacks.rb:25:in `on_file_autoloaded'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:28:in `require'
/opt/redmine/plugins/redmine_local_avatars/init.rb:49:in `<top (required)>'
/opt/redmine/lib/redmine/plugin_loader.rb:31:in `load'
/opt/redmine/lib/redmine/plugin_loader.rb:31:in `run_initializer'
/opt/redmine/lib/redmine/plugin_loader.rb:108:in `each'
/opt/redmine/lib/redmine/plugin_loader.rb:108:in `block in load'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:427:in `instance_exec'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:427:in `block in make_lambda'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:604:in `block (2 levels) in default_terminator'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:603:in `catch'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:603:in `block in default_terminator'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:199:in `block in halting'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `block in invoke_before'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `each'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `invoke_before'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:105:in `run_callbacks'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/reloader.rb:88:in `prepare!'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/application/finisher.rb:124:in `block in <module:Finisher>'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/initializable.rb:32:in `instance_exec'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/initializable.rb:32:in `run'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/initializable.rb:61:in `block in run_initializers'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/initializable.rb:60:in `run_initializers'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/application.rb:391:in `initialize!'
/opt/redmine/config/environment.rb:16:in `<top (required)>'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/dependencies.rb:332:in `block in require'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/dependencies.rb:299:in `load_dependency'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.6/lib/active_support/dependencies.rb:332:in `require'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/application.rb:367:in `require_environment!'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/application.rb:533:in `block in run_tasks_blocks'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:243:in `block in invoke_prerequisites'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:241:in `each'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:241:in `invoke_prerequisites'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:218:in `block in invoke_with_call_chain'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/commands/rake/rake_command.rb:24:in `block (2 levels) in perform'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/commands/rake/rake_command.rb:24:in `block in perform'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/lib/rake/rake_module.rb:59:in `with_application'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/commands/rake/rake_command.rb:18:in `perform'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/command.rb:50:in `invoke'
/opt/redmine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.6/lib/rails/commands.rb:18:in `<top (required)>'
Tasks: TOP => redmine:plugins:migrate => environment
(See full trace by running task with --trace)
```

**Comment by ashrafalzyoud** [comment]
@conathan 
@Kamp11 
@draschba 
@dup2 
@giraypultar 
can u update it in redmine 5??

**Comment by ashrafalzyoud** [comment]
@adrianobr

**Comment by conathan** [comment]
This is what we did (made from diff -Naurp, so doesn't show renames very well)

The Zeitwerk::NameError message, is in regards to the name/path of the file, not matching what is being defined in it (which is why I did the renames).   I don't know the maintainer plans to rename the files, or change the class names as a warning (may not match with what the maintainer will do).

Note: I got nothing to do with the project, and we are running it successfully under Redmine 5

[redmine_local_avatars.bravenet.patch.txt](https://github.com/ncoders/redmine_local_avatars/files/9271469/redmine_local_avatars.bravenet.patch.txt)

**Comment by ashrafalzyoud** [comment]
@conathan 
if im assigne group
there no avatar
how i can do that in this plugin

**Comment by conathan** [comment]
I don't quite follow the question,  but in general to change your avatar if this is extracted to the plugins folder, you can go to your user page (Click the loginname up top where it says Logged in as 'conathan'), click on Avatar in the General/Group/Projects/Avatar tab, Browse, then click Save.

ex:/
redmine.example.com/users/1234/edit?tab=avatar

And then it shows up anywhere an avatar for your user would be used.
* Redmine supports avatars without this, but depends on the gravatar service (personally never looked into gravatar, but we wanted to do avatars locally)

**Comment by ashrafalzyoud** [comment]
thx for youe help and support
this plugin support avatar for users only
for author and assigne 
but if im assigne to group (group 1)
im not found any way to put avatar for groups??

**Comment by conathan** [comment]
I don't think I can help you there, apologies.

**Comment by dup2** [comment]
Should be fixes with #33
